docs/refguide/action.md

```diff
@@ -1,79 +1,422 @@
+class Doubler {
+    value = 0
 
-## Bound actions
+    constructor(value) {
+        makeAutoObservable(this)
+    }
 
-The `action` decorator / function follows the normal rules for binding in javascript.
-However, `action.bound` can be used to automatically bind actions to the targeted object.
-Note that unlike `action`, `(@)action.bound` does not take a name parameter, so the name will always be based on the property name to which the action is bound.
+    increment() {
+        this.value++
+        this.value++
+    }
+}
+```
 
-Example:
+<!--action.bound-->
 
 ```javascript
-class Ticker {
-    @observable tick = 0
+import { makeObservable, observable, computed, action } from "mobx"
+
+class Doubler {
+    value = 0
+
+    constructor(value) {
+        makeObservable(this, {
+            value: observable,
+            increment: action.bound
+        })
+    }
 
-    @action.bound
     increment() {
-        this.tick++ // 'this' will always be correct
+        this.value++
+        this.value++
     }
 }
 
-const ticker = new Ticker()
-setInterval(ticker.increment, 1000)
+const doubler = new Doubler()
+setInterval(doubler.increment, 1000) // calling increment this way is safe as it is bound already
+```
+
+<!--action(fn)-->
+
+```javascript
+import { observable, action } from "mobx"
+
+const state = observable({ value: 0 })
+
+const increment = action(state => {
+    state.value++
+    state.value++
+})
+
+increment(state)
+```
+
+<!--runInAction(fn)-->
+
+```javascript
+import { observable } from "mobx"
+
+const state = observable({ value: 0 })
+
+runInAction(() => {
+    state.value++
+    state.value++
+})
+```
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+### `action` can wrap functions
+
+To leverage the transactional nature of MobX as much as possible, actions should be passed as far outward as possible. It is good to mark a class method as action if it modifies state. It is even better to mark event handlers as action; as it is the outer-most transaction that counts. A single unmarked event handler that calls two actions subsequently would still generate two transactions.
+
+To help creating action based event handlers, `action` is not only an annotation, but also a higher order function; it can be called with a function as argument and in that case it will return an `action` wrapped function with the same signature.
+
+For example in React an `onClick` handler can be wrapped like:
+
+```javascript
+const ResetButton = ({ formState }) => (
+    <button
+        onClick={action(e => {
+            formState.resetPendingUploads()
+            formState.resetValues()
+            e.stopPropagation()
+        })}
+    >
+        Reset form
+    </button>
+)
 ```
 
-_Note: don't use *action.bound* with arrow functions; arrow functions are already bound and cannot be rebound._
+For debugging purposes we recommend either name the wrapped function, or pass a name as first argument to `action`.
+
+<div class="detail">
```

Comment:
<img width="833" alt="Screen Shot 2020-09-16 at 1 04 17 PM" src="https://user-images.githubusercontent.com/1010084/93375052-2feca780-f81d-11ea-92e7-38d817cd1b19.png">

I was expecting this to also be inside a `<detail>` element.